### PR TITLE
User can exit

### PR DESCRIPTION
**Pull request by lottetreg**
If the user enters 'exit' at any point the game will stop.

I've also changed some of the integration tests to have the player exit at the end, instead of forcing the user to lose the game in order to stop it.

Allowing the user to exit via a command was not a story, but as I was starting to work on making the number of bombs dynamic it was getting really annoying to keep forcing the player to lose in the integration tests. Thank you @Thomascountz for the suggestion [here](https://github.com/lottetreg/minesweeper/pull/14#discussion_r295490223)!

### CODE REVIEW

**Review by ecmendenhall — APPROVED** [review]
Good idea.

File: lib/game.ex [hunk 1]
```diff
@@ -62,7 +80,20 @@ defmodule Game do
...
-  defp get_move(reader) do
-    Move.translate(reader.read)
+  defp get_input(reader) do
+    reader.read()
+    |> check_for_exit_command()
+  end
+
+  defp check_for_exit_command(input) do
+    if input == @exit_command do
+      {:error, :exit}
```
Comment (Thomascountz) [inline lib/game.ex:90]: I'm curious about the use of an `:error` tuple. The way I understand them is that they denote an error state, but would exiting the game be considered an error? I could be totally off base, but I'm curious!

**Review by Thomascountz — APPROVED** [review]
LGTM!

File: lib/game.ex (hunk 1 above)
Comment (lottetreg) [reply lib/game.ex:90]: Yeah, I had the same thought when I added that, but I couldn't decide on another name. I'll think about what to change it to... Thank you!

**Pull request merged**